Projects/Shapes/Shapes/Shapes/ShapeBuilder.cpp

```diff
@@ -0,0 +1,369 @@
+{
+	system("CLS");
+
+	for (int i = 0; i < Array.GetSize(); i++)
+	{
+		std::cout << "Shape " << i + 1 << std::endl;
+		std::cout << "Area: " << Array[i]->GetArea() << std::endl;
+		std::cout << "Perimeter: " << Array[i]->GetPerimeter() << std::endl;
+	}
+	RequestEnterFromInput();
+}
+
+//-------------------------------------------------------------------------------------------
+void FShapeBuilder::DeleteArray(TDynamicArray<FShape*>& Array) const
+{
+	for (int i = 0; i < Array.GetSize(); i++)
+	{
+		delete Array[i];
+		Array[i] = nullptr;
+	}
+}
+
+
+//-------------------------------------------------------------------------------------------
+//									UTILITY FUNCTIONS
+//-------------------------------------------------------------------------------------------
+FShape* FShapeBuilder::CreateShape() const
+{
+	std::cout << "Choose your Shape!" << std::endl;
+	std::cout << "1 - Circle" << std::endl;
+	std::cout << "2 - Square" << std::endl;
+
+	int ShapeChosen = GetValidIntInput(1, 2);
+
+	if (ShapeChosen == 1)
+	{
+		system("CLS");
+		std::cout << "You chose a circle!" << std::endl;
+		std::cout << "Type in its radius." << std::endl;
+		float Radius = GetValidFloatInput();
+
+		FCircle* Circle = new FCircle(Radius);
+
+		return Circle;
+	}
+	else
+	{
+		system("CLS");
+		std::cout << "You chose a square!" << std::endl;
+		std::cout << "Type in its side length." << std::endl;
+		float Side = GetValidFloatInput();
+
+		FSquare* Square = new FSquare(Side);
+
+		return Square;
+	}
+}
+
+//-------------------------------------------------------------------------------------------
+void FShapeBuilder::RequestEnterFromInput() const
+{
+	std::cin.ignore(INT_MAX, '\n');
+}
+
+//-------------------------------------------------------------------------------------------
+float FShapeBuilder::GetValidFloatInput() const
+{
+	float Option;
+	std::cin >> Option;
+	while (std::cin.fail())
+	{
+		std::cin.clear();
+		RequestEnterFromInput();
+		std::cout << "That is not a valid double. Please try again." << std::endl;
+		std::cin >> Option;
+	}
+	return Option;
+}
+
+//-------------------------------------------------------------------------------------------
+int FShapeBuilder::GetValidIntInput() const
+{
+	int Option;
+	std::cin >> Option;
+	while (std::cin.fail())
+	{
+		std::cin.clear();
+		RequestEnterFromInput();
+		std::cout << "That is not a valid integer. Please try again." << std::endl;
+		std::cin >> Option;
+	}
+	return Option;
+}
+
+//-------------------------------------------------------------------------------------------
+int FShapeBuilder::GetValidIntInput(const int LowerBound, const int UpperBound) const
+{
+	int Option;
+	std::cin >> Option;
+	while (std::cin.fail() || Option < LowerBound || Option > UpperBound)
+	{
+		std::cin.clear();
+		RequestEnterFromInput();
+		std::cout << "That is not a valid integer or within the intended range. Please try again." << std::endl;
+		std::cin >> Option;
+	}
+	return Option;
+}
+
+
```

Comment:
the functions print or create shape are doing the same, you could merge them in just one function, you should create the `for` inside the ask" type" array method and you don't need to overlap the function